crates/sui-json-rpc/src/name_service.rs

```diff
@@ -154,12 +160,59 @@ impl FromStr for Domain {
             return Err(DomainParseError::LabelsEmpty);
         }
 
+        add_tld_if_missing(&mut labels);
+
         let labels = labels.into_iter().map(ToOwned::to_owned).collect();
 
         Ok(Domain { labels })
     }
 }
 
+// Parses a separator from the domain string input.
```

Comment:
Use triple `/` to make it a source code doc.
```suggestion
/// Parses a separator from the domain string input.
```